pkg/controller/certificaterequests/controller.go

```diff
@@ -61,6 +62,7 @@ type Controller struct {
 	fieldManager string
 
 	certificateRequestLister cmlisters.CertificateRequestLister
+	secretLister             corelisters.SecretLister
```

Comment:
Suggestion: add a comment ( i.e along the lines of 'we need to wait for `Secret`s to be synced to avoid a situation where CA issuer's `Secret` is not yet in cached at a time when issuance is attempted`) and linking https://github.com/cert-manager/cert-manager/issues/5216 explaining why we need to wait